drivers/temperature/max31865pmb1/max31865.c

```diff
@@ -0,0 +1,424 @@
+	ret = max31865_write(device, MAX31865_HFAULTLSB_REG, (uint8_t)lsb);
+	if(ret)
+		return ret;
+
+	lsb = no_os_field_get(0x00FF, lower);
+	msb = no_os_field_get(0xFF00, lower);
+
+	ret = max31865_write(device, MAX31865_LFAULTMSB_REG, (uint8_t)msb);
+	if(ret)
+		return ret;
+
+	return max31865_write(device, MAX31865_LFAULTLSB_REG, (uint8_t)lsb);
+}
+
+/**
+ * @brief Read the raw 16-bit lower threshold value
+ * @param device MAX31865 descriptor
+ * @param low_threshold pointer to hold the 16-bit lower threshold value
+ * @return 0 in case of success, negative error code otherwise
+ */
+int max31865_get_lower_threshold(struct max31865_dev *device,
+				 uint16_t *low_threshold)
+{
+	uint8_t low_msb;
+	uint8_t low_lsb;
+	int ret;
+
+	ret = max31865_read(device, MAX31865_LFAULTMSB_REG, &low_msb);
+	if(ret)
+		return ret;
+
+	ret = max31865_read(device, MAX31865_LFAULTLSB_REG, &low_lsb);
+	if(ret)
+		return ret;
+
+	*low_threshold = (((uint16_t)low_msb << 8) | (uint16_t)low_lsb);
+
+	return 0;
+}
+
+/**
+ * @brief Read the raw 16-bit upper threshold value
+ * @param device MAX31865 descriptor
+ * @param up_threshold pointer to hold the 16-bit upper threshold value
+ * @return 0 in case of success, negative error code otherwise
+ */
+int max31865_get_upper_threshold(struct max31865_dev *device,
+				 uint16_t *up_threshold)
+{
+	uint8_t high_msb;
+	uint8_t high_lsb;
+	int ret;
+
+	ret = max31865_read(device, MAX31865_HFAULTMSB_REG, &high_msb);
+	if(ret)
+		return ret;
+
+	ret = max31865_read(device, MAX31865_HFAULTLSB_REG, &high_lsb);
+	if(ret)
+		return ret;
+
+	*up_threshold = (((uint16_t)high_msb << 8) | (uint16_t)high_lsb);
+
+	return 0;
+}
+
+/**
+ * @brief N-wire option for RTD measurement setup
+ * @param device MAX31865 descriptor
+ * @param is_odd_wire If true, uses 3-wire setup. Else, uses 2/4-wire setup.
+ * @return 0 in case of success, negative error code otherwise
+ */
+int max31865_set_wires(struct max31865_dev *device, bool is_odd_wire)
+{
+	uint8_t reg_data = 0x00;
+	int ret;
+
+	ret = max31865_read(device, MAX31865_CONFIG_REG, &reg_data);
+	if(ret)
+		return ret;
+
+	if (is_odd_wire)
+		reg_data |= MAX31865_CONFIG_3WIRE;
+	else
+		reg_data &= MAX31865_CONFIG_2_4WIRE;
+
+	return max31865_write(device, MAX31865_CONFIG_REG, reg_data);
+}
+
+
+/**
+ * @brief Read the raw 16-bit value from the RTD_REG in one shot mode
+ * @param device MAX31865 descriptor
+ * @param rtd_reg pointer to hold the 16-bit raw RTD_REG value
+ * @return 0 in case of success, negative error code otherwise
+ */
+int max31865_read_rtd(struct max31865_dev *device, uint16_t *rtd_reg)
+{
+	uint8_t reg_data;
+	int ret;
+
+	ret = max31865_clear_fault(device);
+	if(ret)
+		return ret;
+
+	ret = max31865_enable_bias(device, true);
+	if(ret)
+		return ret;
+
+	ret = max31865_read(device, MAX31865_CONFIG_REG, &reg_data);
+	if(ret)
+		return ret;
+
+	reg_data |= MAX31865_CONFIG_1SHOT;
+
+	ret = max31865_write(device, MAX31865_CONFIG_REG, reg_data);
+	if(ret)
+		return ret;
+
```

Comment:
You could improve this by implementing an `reg_update()` function. There are a few other places where this would be the case.

Reply:
Hi Ciprian,
Does this mean that I need to implement a "max31865_reg_update()" function?